iceoryx2-cli/iox2-service/src/command_subscribe.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::cli::{DataRepresentation, SubscribeOptions};
+use anyhow::anyhow;
+use anyhow::Result;
+use iceoryx2::prelude::*;
+use iceoryx2::sample::Sample;
+use iceoryx2::service::builder::{CustomHeaderMarker, CustomPayloadMarker};
+use iceoryx2::service::static_config::message_type_details::TypeDetail;
+use iceoryx2_cli::Format;
+use std::fs::File;
+use std::io::Write;
+use std::time::Duration;
+use std::time::Instant;
+
+fn raw_data_to_hex_string(raw_data: &[u8]) -> String {
+    let mut ret_val = String::with_capacity(2 * raw_data.len());
+    for byte in raw_data {
+        ret_val.push_str(&format!("{byte:0>2x} "));
+    }
+
+    ret_val
```

Comment:
This is a bit more efficient. The string will already have the right size since 3 ascii chars will be added, due to the whitespace and also the `write` macro writes directly into the buffer without allocating, unlike the `format` macro.
```suggestion
    let mut ret_val = String::with_capacity(raw_data.len() * 3);
    for byte in raw_data {
        write!(&mut ret_val, "{byte:02x} ").unwrap();
    }

    ret_val
```

One could also think of getting rid of the whitespace at the end by using `" {byte:02x}"` and not adding a whitespace at the place where the string is used